src/storage/checkpoints/checkpoint.cpp

```diff
@@ -0,0 +1,156 @@
+#include "storage/checkpoints/checkpoint.h"
+#include <common/worker_pool.h>
+#include <storage/block_compactor.h>
+#include <storage/recovery/disk_log_provider.h>
+#include <fstream>
+#include "common/constants.h"
+namespace terrier::storage {
+
+bool Checkpoint::TakeCheckpoint(const std::string &path, catalog::db_oid_t db, const char *cur_log_file,
+                                uint32_t num_threads, common::WorkerPool *thread_pool_) {
+  // get db catalog accessor
+  auto txn = txn_manager_->BeginTransaction();
+  auto accessor = catalog_->GetAccessor(static_cast<common::ManagedPointer<transaction::TransactionContext>>(txn), db);
+  std::unordered_set<catalog::table_oid_t> table_oids = accessor->GetAllTableOids();
+
+  // lock the log_serializer buffer to prevent further logging
+  common::ManagedPointer<LogSerializerTask> log_serializer_task = log_manager_->log_serializer_task_;
+  log_serializer_task->flush_queue_latch_.Lock();
+
+  // copy all data tables to serialize later
+  for (const auto &oid : table_oids) {
+    // copy data table
+    common::ManagedPointer<storage::SqlTable> curr_sql_table = accessor->GetTable(oid);
+    storage::DataTable *curr_data_table = curr_sql_table->table_.data_table_;
+    queue.emplace_back(oid, curr_data_table);
+  }
+  std::string tmp = "_catalog";
+  std::string catalog_file = std::string(cur_log_file) + tmp; //catalog file: test.log_catalog
+  log_serializer_task->flush_queue_latch_.Unlock();
+
+  // filter catalog logs to catalog file
+  FilterCatalogLogs(cur_log_file, catalog_file);
+  remove(cur_log_file);
+
+  // create empty log file for other logs
+  std::ofstream new_log_file(cur_log_file, std::ios::in | std::ios::app);
+  log_manager_->ResetLogFilePath(cur_log_file);
+
+  auto workload = [&](uint32_t worker_id) {
+    // copy contents of table to disk
+    WriteToDisk(path, accessor, db);
+  };
+  for (auto i = 0u; i < num_threads; i++) {
+    thread_pool_->SubmitTask([i, &workload] { workload(i); });
+  }
+  thread_pool_->WaitUntilAllFinished();
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+  if (queue.size() > 0) {
+    // the table oid that failed to be backup
```

Comment:
I don't understand this comment because usually I would take it to understand that you are returning the table oid that failed to be backed up, but here you are returning true/false. So do you mean a table oid failed to be backed up? (not a specific table?)